precompiles/avs/tx.go

```diff
@@ -305,5 +322,79 @@ func (p Precompile) RegisterBLSPublicKey(
 		return nil, err
 	}
 
+	if err = p.EmitPublicKeyRegistered(ctx, stateDB, blsParams); err != nil {
+		return nil, err
+	}
+	return method.Outputs.Pack(true)
+}
+
+// OperatorSubmitTask operator submit results
+func (p Precompile) OperatorSubmitTask(
+	ctx sdk.Context,
+	_ common.Address,
+	_ *vm.Contract,
+	stateDB vm.StateDB,
+	method *abi.Method,
+	args []interface{},
+) ([]byte, error) {
+	if len(args) != len(p.ABI.Methods[MethodOperatorSubmitTask].Inputs) {
+		return nil, fmt.Errorf(cmn.ErrInvalidNumberOfArgs, len(p.ABI.Methods[MethodOperatorSubmitTask].Inputs), len(args))
+	}
+	resultParams := &avstypes.TaskResultParams{}
+
+	callerAddress, ok := args[0].(common.Address)
+	if !ok || (callerAddress == common.Address{}) {
+		return nil, fmt.Errorf(exocmn.ErrContractInputParaOrType, 0, "common.Address", callerAddress)
+	}
+	resultParams.CallerAddress = callerAddress[:]
+
+	taskID, ok := args[1].(uint64)
+	if !ok {
+		return nil, fmt.Errorf(exocmn.ErrContractInputParaOrType, 1, "uint64", args[1])
+	}
+	resultParams.TaskID = taskID
+
+	taskResponse, ok := args[2].([]byte)
+	if !ok {
+		return nil, fmt.Errorf(exocmn.ErrContractInputParaOrType, 2, "[]byte", taskResponse)
+	}
+	resultParams.TaskResponse = taskResponse
+
+	blsSignature, ok := args[3].([]byte)
+	if !ok {
+		return nil, fmt.Errorf(exocmn.ErrContractInputParaOrType, 3, "[]byte", blsSignature)
+	}
+	resultParams.BlsSignature = blsSignature
+
+	taskAddr, ok := args[4].(common.Address)
+	if !ok || (taskAddr == common.Address{}) {
+		return nil, fmt.Errorf(exocmn.ErrContractInputParaOrType, 4, "common.Address", taskAddr)
+	}
+	resultParams.TaskContractAddress = taskAddr
+
+	phase, ok := args[5].(uint8)
+	if !ok {
```

Comment:
No validation of the `phase` number? Only certain values are supported right?

Reply:
fixed